code/controllers/subsystem/time_track.dm

```diff
@@ -16,33 +17,81 @@ SUBSYSTEM_DEF(time_track)
-		last_measurement = 0
-		var/current_realtime = REALTIMEOFDAY
-		var/current_byondtime = world.time
-		var/current_tickcount = world.time/world.tick_lag
-
-		if (!first_run)
-			var/tick_drift = max(0, (((current_realtime - last_tick_realtime) - (current_byondtime - last_tick_byond_time)) / world.tick_lag))
-
-			time_dilation_current = tick_drift / (current_tickcount - last_tick_tickcount) * 100
-
-			time_dilation_avg_fast = MC_AVERAGE_FAST(time_dilation_avg_fast, time_dilation_current)
-			time_dilation_avg = MC_AVERAGE(time_dilation_avg, time_dilation_avg_fast)
-			time_dilation_avg_slow = MC_AVERAGE_SLOW(time_dilation_avg_slow, time_dilation_avg)
-		else
-			first_run = FALSE
-		last_tick_realtime = current_realtime
-		last_tick_byond_time = current_byondtime
-		last_tick_tickcount = current_tickcount
-		SSblackbox.record_feedback("associative", "time_dilation_current", 1, list("[SQLtime()]" = list("current" = "[time_dilation_current]", "avg_fast" = "[time_dilation_avg_fast]", "avg" = "[time_dilation_avg]", "avg_slow" = "[time_dilation_avg_slow]")))
-		time_dilation_text = "Time Dilation: [round(time_dilation_current,1)]% AVG:([round(time_dilation_avg_fast,1)]%, [round(time_dilation_avg,1)]%, [round(time_dilation_avg_slow,1)]%)"
+
+	var/current_realtime = REALTIMEOFDAY
+	var/current_byondtime = world.time
+	var/current_tickcount = world.time/world.tick_lag
+	GLOB.glide_size_multiplier = (current_byondtime - last_tick_byond_time) / (current_realtime - last_tick_realtime)
+
+	if(times_fired % 10)	// everything else is once every 10 seconds
+		return
+
+	if (!first_run)
+		var/tick_drift = max(0, (((current_realtime - last_tick_realtime) - (current_byondtime - last_tick_byond_time)) / world.tick_lag))
+
+		time_dilation_current = tick_drift / (current_tickcount - last_tick_tickcount) * 100
+
+		time_dilation_avg_fast = MC_AVERAGE_FAST(time_dilation_avg_fast, time_dilation_current)
+		time_dilation_avg = MC_AVERAGE(time_dilation_avg, time_dilation_avg_fast)
+		time_dilation_avg_slow = MC_AVERAGE_SLOW(time_dilation_avg_slow, time_dilation_avg)
+	else
+		first_run = FALSE
+	last_tick_realtime = current_realtime
+	last_tick_byond_time = current_byondtime
+	last_tick_tickcount = current_tickcount
+	SSblackbox.record_feedback("associative", "time_dilation_current", 1, list("[SQLtime()]" = list("current" = "[time_dilation_current]", "avg_fast" = "[time_dilation_avg_fast]", "avg" = "[time_dilation_avg]", "avg_slow" = "[time_dilation_avg_slow]")))
+	log_perf(
+		list(
+			world.time,
+			length(GLOB.clients),
+			time_dilation_current,
+			time_dilation_avg_fast,
+			time_dilation_avg,
+			time_dilation_avg_slow,
+			MAPTICK_LAST_INTERNAL_TICK_USAGE,
+			length(SStimer.timer_id_dict),
+			SSair.cost_turfs,
+			SSair.cost_groups,
+			SSair.cost_highpressure,
+			SSair.cost_hotspots,
+			SSair.cost_superconductivity,
+			SSair.cost_pipenets,
+			SSair.cost_rebuilds,
+			SSair.get_active_turfs(), //does not return a list, which is what we want
```

Comment:
```suggestion
```
These removed entries may need to be replaced with some of the new procs/vars, like `turf_process_time()`, `heat_process_time()`, `cost_post_process`, etc., depending on how logs are formatted.